pkg/validation/spec/swagger.go

```diff
@@ -46,6 +46,21 @@ func (s Swagger) MarshalJSON() ([]byte, error) {
 	return swag.ConcatJSON(b1, b2), nil
 }
 
+func (s Swagger) MarshalNext() ([]byte, error) {
+	return jsonv2.Marshal(s)
+}
```

Comment:
is there a reason not to follow the same pattern as Unmarshal, like 

```
func (s Swagger) MarshalJSON() ([]byte, error) {
	if internal.UseOptimizedJSONMarshaling {
		return jsonv2.Marshal(s)
	}
  ...
}
```

then all the call-sites could switch to calling `Swagger#MarshalJSON()` rather than calling `MarshalNext()` directly, which I assume we don't really want long-term

Reply:
Done!